### ISSUE TITLE
pad fails, takes amount from unrelated account into account

### ISSUE THREAD

**Issue by blais** [beancount/beancount#362]
**[Original report](https://bitbucket.org/blais/beancount/issue/362) by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

I have account `Assets:Bank` and `Assets:Bank-Two`: a `pad` statement for `Assets:Bank` also takes the amount from `Assets:Bank-Two`, leading to the wrong balance.

```
1970-01-01 open Assets:Bank
1970-01-01 open Assets:Bank-Two
1970-01-01 open Equity:Opening-Balance
1970-01-01 open Equity:Adjustments
1970-01-01 open Expenses:Food

2019-01-01 * "Opening balance"
  Assets:Bank                  20.00 GBP
  Equity:Opening-Balance      -20.00 GBP

2019-01-03 * "Tesco" "Buy food"
  Expenses:Food                30.00 GBP
  Assets:Bank                 -30.00 GBP

2019-01-04 balance Assets:Bank -10.00 GBP

2019-01-03 * "Tesco" "Buy food"
  Expenses:Food                 6.00 GBP
  Assets:Bank-Two              -6.00 GBP

; Get rid of overdraft somehow and get a £50 balance

2019-01-04 pad Assets:Bank Equity:Adjustments
2019-01-05 balance Assets:Bank 50.00 GBP
```

```
bean-query ~/t.beancount "Select * where account ~ 'Bank$'"
/home/tbm/t.beancount:25:      Balance failed for 'Assets:Bank': expected 50.00 GBP != accumulated 56.00 GBP (6.00 too much)

   2019-01-05 balance Assets:Bank                                     50.00 GBP


   date    f payee                              narration                                position 
---------- - ----- -------------------------------------------------------------------- ----------
2019-01-01 *       Opening balance                                                       20.00 GBP
2019-01-03 * Tesco Buy food                                                             -30.00 GBP
2019-01-04 P       (Padding inserted for Balance of 50.00 GBP for difference 66.00 GBP)  66.00 GBP
```

To go from -10 to +50, we should pad by 60.  The 6 comes from `Assets:Bank-Two`, which is completely unrelated.

**Comment by blais** [comment]
**Original comment by Martin Michlmayr (Bitbucket: [tbm13](https://bitbucket.org/tbm13), GitHub: [tbm](https://github.com/tbm)).**

----------------------------------------

If you delete the unrelated transaction involving `Assets:Bank-Two`, then it will work.

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

That's crazy!

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Holy cow!
https://bitbucket.org/blais/beancount/src/be8b5d4d50717a138a5a4013e6cd4645b861a04f/beancount/ops/pad.py#lines-64
https://bitbucket.org/blais/beancount/src/be8b5d4d50717a138a5a4013e6cd4645b861a04f/beancount/core/account.py#lines-189

**Comment by blais** [comment]
**Original comment by Martin Blais (Bitbucket: [blais](https://bitbucket.org/blais), GitHub: [blais](https://github.com/blais)).**

----------------------------------------

Fixed #362: Predicate matching a parent account was matching accounts with word breaks but not at the boundary of an account name.
